src/System.Private.Xml.Linq/tests/TreeManipulation/SaveWithFileName.cs

```diff
@@ -0,0 +1,287 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using CoreXml.Test.XLinq;
+using Microsoft.Test.ModuleCore;
+using XmlCoreTest.Common;
+using Xunit;
+
+namespace XLinqTests
+{
+    public class SaveWithFileName
+    {
+        private string _fileName = "SaveBaseline";
+
+        [Fact]
+        public void XDocumentSaveToFile()
+        {
+            SerializeWithSaveOptions(SerializeXDocumentSaveToFile, false, false, true);
+        }
+
+        [Fact]
+        public void XDocumentSave()
+        {
+            string markup = "<e> <e2 /> </e>";
+            try
+            {
+                XDocument d = XDocument.Parse(markup, LoadOptions.PreserveWhitespace);
+                d.Save(_fileName);
+            }
+            finally
+            {
+                Assert.True(System.IO.File.Exists(_fileName));
+                Assert.Equal("<?xml version=\"1.0\" encoding=\"utf-8\"?>\r\n" + markup, File.ReadAllText(_fileName));
```

Comment:
XmlWriterSettings.NewLineChars slooks like it's initialized to Environment.NewLine. So I'm guessing it will vary

---
In reply to: [85858863](https://github.com/dotnet/corefx/pull/13143#discussion_r85858863) [](ancestors = 85858863)